scripts/setup_truefoundry_cluster.sh

```diff
@@ -87,23 +87,48 @@ function make_request() {
     return 0
 }
 
+
+function is_cluster_provisioned() {
+    log_info "is_cluster_provisioned: Checking if cluster exists and is provisioned..."
+
+    # Make GET request to check cluster status
+    local response=$(make_request "GET" "${CONTROL_PLANE_URL}/api/svc/v1/cluster/${CLUSTER_NAME}" "" "200") || {
+        log_info "is_cluster_provisioned: Got error while querying for cluster ${CLUSTER_NAME}"
```

Comment:
```suggestion
        log_error "is_cluster_provisioned: Got error while querying for cluster ${CLUSTER_NAME}"
```

Reply:
log_error would also `return 1` which would exit from TF since data external module doesn't like any non-zero return code. 

am updating the log message

Reply:
we want to achieve. 

if cluster doesn't exist. 
setup_provider  + create cluster

if cluster exists,  just fetch cluster_id

Reply:
What is the problem if it fails ? that should be ideal behaviour - the terraform should fail

Reply:
we don't want the script to fail if the cluster doesn't exists.